hack/run-e2e.sh

```diff
@@ -87,8 +87,8 @@ function evaluate_test_num {
 }
 
 # Default versions k8s and kind
-K8S_VERSION=${K8S_VERSION:-v1.29.2}
-KIND_VERSION=${KIND_VERSION:-v0.22.0}
+K8S_VERSION=${K8S_VERSION:-v1.34.0}
+KIND_VERSION=${KIND_VERSION:-v0.23.0}
```

Comment:
I'm seeing `v0.30.0` as the latest kind release, ref: https://github.com/kubernetes-sigs/kind/releases